torch/csrc/jit/passes/frozen_linear_transpose.cpp

```diff
@@ -0,0 +1,111 @@
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/ir/ir_views.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/frozen_linear_transpose.h>
+#include <torch/csrc/jit/passes/utils/optimization_utils.h>
+#include <torch/csrc/jit/runtime/graph_executor.h>
+#include <iostream>
+
+namespace torch {
+namespace jit {
+namespace {
+
+using Tensor = at::Tensor;
+
+class TransposeFrozenLinear {
+ public:
+  TransposeFrozenLinear(std::shared_ptr<Graph> graph)
+      : graph_(std::move(graph)) {}
+
+  bool run() {
+    handleBlockAndSubblocks(graph_->block());
+    return graph_modified;
+  }
+
+  bool is_constant_linear_op(Node* node) {
+    if (node->kind() != aten::linear) {
+      return false;
+    }
+    auto weight = node->namedInput("weight");
+    if (weight->type() == NoneType::get()) {
+      return false;
+    }
+
+    // This also filters out out-variants of the linear op.
+    if (nonConstantParameters(node)) {
+      return false;
+    }
+
+    // Op is only profitable on CUDA
+    auto weight_tensor = constant_as<Tensor>(weight).value();
```

Comment:
I thought our findings were that this was profitable on CPU as well ?

Reply:
As 1CPU is the thing  we care about, I have updated it to also optimize for the CPU case.